### PR TITLE
Feature/spring cloud autoconfig init

### PR DESCRIPTION
**Pull request by moarychan**
Migrate repo:
https://github.com/microsoft/spring-cloud-azure/tree/master/spring-cloud-azure-storage
https://github.com/microsoft/spring-cloud-azure/tree/master/spring-cloud-azure-autoconfigure

### CODE REVIEW

**Comment by moarychan** [comment]
@saragluna , please help review

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -88,9 +93,12 @@ org.springframework.boot:spring-boot-starter;2.3.0.RELEASE
 org.springframework.boot:spring-boot;2.3.0.RELEASE
 org.springframework.boot:spring-boot-starter-parent;2.3.0.RELEASE
 org.springframework.data:spring-data-commons;2.3.0.RELEASE
+org.springframework.data:spring-data-redis;2.2.5.RELEASE
+org.springframework.data:spring-data-mongodb;2.2.5.RELEASE
```
Comment (saragluna) [inline eng/versioning/external_dependencies.txt:97]: Are these two spring-data-* versions align with spring-data 2.3.0?

File: sdk/spring/azure-spring-cloud-autoconfigure/pom.xml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                <include>org.springframework:spring-context-support:[5.2.6.RELEASE]</include> <!-- {x-include-update;org.springframework:spring-context-support;external_dependency} -->
+                <include>com.microsoft.azure:spring-cloud-azure-context:[1.2.8-beta.1]</include> <!-- {x-include-update;com.microsoft.azure:spring-cloud-azure-context;current} -->
+                <include>org.springframework:spring-jms:[5.2.6.RELEASE]</include> <!-- {x-include-update;org.springframework:spring-jms;external_dependency} -->
+                <include>com.fasterxml.jackson.core:jackson-core:[2.10.1]</include> <!-- {x-include-update;com.fasterxml.jackson.core:jackson-core;external_dependency} -->
+                <include>org.hibernate.validator:hibernate-validator:[6.1.5.Final]</include> <!-- {x-include-update;org.hibernate.validator:hibernate-validator;external_dependency} -->
+                <include>org.apache.qpid:qpid-jms-client:[0.43.0]</include> <!-- {x-include-update;org.apache.qpid:qpid-jms-client;external_dependency} -->
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-autoconfigure/pom.xml:238]: We should sort these lines.

**Comment by saragluna** [comment]
Including these two modules in this issue: https://github.com/Azure/azure-sdk-for-java/issues/13523

File: sdk/spring/azure-spring-cloud-autoconfigure/pom.xml (hunk 2 above)
Comment (moarychan) [reply sdk/spring/azure-spring-cloud-autoconfigure/pom.xml]: Done, the same operation for integration modules.

File: eng/versioning/external_dependencies.txt (hunk 1 above)
Comment (moarychan) [reply eng/versioning/external_dependencies.txt]: Just discussed, we do not change it first.

**Pull request merged**